ale_linters/slim/slimlint.vim

```diff
@@ -1,5 +1,19 @@
 " Author: Markus Doits - https://github.com/doits
-" Description: slim-lint for Slim files, based on hamllint.vim
+" Description: slim-lint for Slim files
+
+function! ale_linters#slim#slimlint#GetCommand(buffer) abort
+    let l:rubocop_config = ale#path#ResolveLocalPath(
+    \   a:buffer,
+    \   '.rubocop.yml',
+    \   ''
+    \)
+
+    if !empty(l:rubocop_config)
+        let $SLIM_LINT_RUBOCOP_CONF = l:rubocop_config
```

Comment:
You should use the `--config` option with the config path instead of changing the environment variable, which has more side-effects. Use `ale#Escape` on the path.

Reply:
It looks like `slim` has separate configuration files for itself, but it also uses rubocop configuration files? I don't know what's going on there.

The danger in setting the environment variable is that will will be applied to other invocations of `slim-lint` where no configuration file is found. I don't think the environment variable will persist after Vim is closed.

Reply:
> It looks like slim has separate configuration files for itself, but it also uses rubocop configuration files? I don't know what's going on there.

It is explained in the linter's readme:

https://github.com/sds/slim-lint/blob/5b258d4c922b47ee28e593992bbdcbdfd47d61fe/lib/slim_lint/linter/README.md#rubocop

I must admit it is a little bit complicated, but the env var seems exactly be the way how slim-lint expects linters to do it ...

Reply:
I wouldn't recommend setting environment variables like this, as you don't know if the environment variable is going to be changed again before the actual command is run. Plus there's the problem of the environment variable being used when you don't want it to be used, where you run `slim-lint` once with a configuration file available, and then again without it.

Reply:
I recommend prefixing the command string with the environment variable instead. Use `X=... <command>` for Unix, and `set X=... && <command>` for Windows. You can use `ale#Has('win32')` so you can test both versions when running the tests on Unix.

Reply:
Look at how `ale#Has` is used in existing Vader tests. There's a `Dictionary` which can be set for effectively mocking `has()` calls.